opentok-whiteboard.js

```diff
@@ -96,90 +93,95 @@ var OpenTokWhiteboard = angular.module('opentok-whiteboard', ['opentok'])
             };
             
             scope.erase = function () {
-                //scope.color = element.css("background-color") || "#fff";
-                scope.lineWidth = 50;
                 scope.erasing = true;
             };
             
             scope.capture = function () {
                 if (iOS) {
                     // On iOS you can put HTML in a mailto: link
-                    window.location.href = "mailto:?subject=Whiteboard&Body=<img src='" + canvas.toDataURL('image/png') + "'>";
+                    $window.location.href = "mailto:?subject=Whiteboard&Body=<img src='" + canvas.toDataURL('image/png') + "'>";
                 } else {
                     // We just open the image in a new window
-                    window.open(canvas.toDataURL('image/png'));
+                    $window.open(canvas.toDataURL('image/png'));
                 }
             };
 
             scope.undo = function () {
                 if (!undoStack.length)
                     return;
-                var undodata = undoStack.pop();
-                undoWhiteBoard(undodata);
-                redoStack.push(undodata);
-                sendUpdate('otWhiteboard_undo', undodata);
+                var data = undoStack[undoStack.length - 1];
+                undoWhiteBoard(data);
+                sendUpdate('otWhiteboard_undo', data);
             };
 
             var undoWhiteBoard = function (data) {
```

Comment:
I see a race condition issue that could happen here. eg.

1. User A is drawing and then clicks the undo button
2. User B starts drawing before he receives the undo message from User A. He will undo what he was drawing instead of what User A intended to undo and now User A and User B will be out of sync.

It will work fine if everyone takes turns drawing and waits enough time in between but if multiple people are drawing and clicking undo then things will go haywire.

I think that you are going to need to do something like search through the drawHistory to find the corresponding data that needs to be hidden.